### PR TITLE
add -pull-timeout flag

### PR DESCRIPTION
**Pull request by darkonie**
add pull timeout flag

### CODE REVIEW

File: api/config.go [hunk 1]
```diff
@@ -105,6 +107,9 @@ func LoadDefaultConfig(args []string) (config Config, err error) {
 	config.FlagPullInterval = 60
 	config.FlagUpdateHealthReportInterval = 60
 
+	// Set default pull timeout to 10 seconds
+	config.FlagPullTimeout = 10
```
Comment (malnick) [inline api/config.go]: Do we have any data to support this timeout on large (greater than 50 nodes) clusters?

**Comment by malnick** [comment]
One comment on the data backing up the chosen timeout. Can we test this on a large cluster to make sure we're doing the right thing?

File: api/config.go (hunk 1 above)
Comment (darkonie) [inline api/config.go]: yeah 10 sec is probably overkill, we might want to set it to 3 sec as it was before

**Comment by darkonie** [comment]
@malnick all tests conducted on a fresh cluster with 100 nodes went green. We should probably set the default timeout to 3 sec and expose a CLI flag to increase the timeout if needed. So we don't have recompile the code in order to bump a timeout

**Comment by malnick** [comment]
Do you want to lop that into this PR or make it separate? I'm a :ship: on this once you update the timeout accordingly.

**Pull request merged**